src/repositories/UniswapV3PoolRepository.ts

```diff
@@ -59,4 +64,60 @@ export class UniswapV3PoolRepository {
 
     return getModelForClass(UniswapV3Pool).find(filter).sort({token0: 1, token1: 1}).exec();
   }
+
+  async findBestPool(props: {
+    protocol: string;
+    fromChain: string[];
```

Comment:
you using `fromChain[0]` so lest make it just `string` and pass `fromChain[0]` as parameter here.